src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -540,11 +532,14 @@ static int uct_rc_mlx5_iface_srq_topo(uct_rc_mlx5_iface_common_t *iface,
 {
     uct_ib_mlx5_md_t *ib_md = ucs_derived_of(md, uct_ib_mlx5_md_t);
 
-    /* Cyclic SRQ is supported with HW TM and DEVX only. */
+    /* Cyclic SRQ is supported with DEVX only if: */
     if (((mlx5_config->srq_topo == UCT_RC_MLX5_SRQ_TOPO_AUTO) ||
         (mlx5_config->srq_topo == UCT_RC_MLX5_SRQ_TOPO_CYCLIC)) &&
-        UCT_RC_MLX5_TM_ENABLED(iface) &&
-        (ib_md->flags & UCT_IB_MLX5_MD_FLAG_DEVX)) {
+        (ib_md->flags & UCT_IB_MLX5_MD_FLAG_DEVX) &&
+        (/* - either HW TM is enabled */
+         UCT_RC_MLX5_TM_ENABLED(iface) ||
+         /* - or DevX RMP object is supported for SW TM */
+         (ib_md->flags & UCT_IB_MLX5_MD_FLAG_RMP))) {
```

Comment:
can we reuse uct_rc_mlx5_is_devx_srq_supported here?

Reply:
BTW, if user selects UCT_RC_MLX5_SRQ_TOPO_CYCLIC, and cyclic is not supported, shouldn't we fail?
@brminich

Reply:
when cyclic is not supported? AFAIU, it is always supported by HW and we check for DEVX flags to make sure we can set it properly.

Reply:
so why do we have config value for auto?
IMO the difference between "cyclic" and "auto" is that "cyclic" would fail if can't be done

Reply:
ic, yes, this would need to be fixed

Reply:
done, now it will fail UCT iface initialization and print the following error:
```
$UCX_IB_MLX5_DEVX_OBJECTS=rcqp,dct UCX_RC_SRQ_TOPO=cyclic UCX_TLS=ib ../install/bin/ucx_info -e -u tr -N 1 -n 1
[1583431045.823286] [hpc-test-node-upstream-01:30789:0]  rc_mlx5_iface.c:542  UCX  ERROR Cyclic SRQ WQ type was requested, but it is not supported by the configuration
<Failed to create UCP worker>
```